include/boost/geometry/algorithms/detail/convex_hull/interface.hpp

```diff
@@ -0,0 +1,394 @@
+                        >(out)), strategy);
+    }
+};
+
+}} // namespace detail::convex_hull
+#endif // DOXYGEN_NO_DETAIL
+
+
+#ifndef DOXYGEN_NO_DISPATCH
+namespace dispatch
+{
+
+
+template
+<
+    typename Geometry,
+    typename Tag = typename tag<Geometry>::type
+>
+struct convex_hull
+    : detail::convex_hull::hull_to_geometry
+{};
+
+template <typename Box>
+struct convex_hull<Box, box_tag>
+{
+    template <typename OutputGeometry, typename Strategy>
+    static inline void apply(Box const& box,
+                             OutputGeometry& out,
+                             Strategy const& )
+    {
+        static bool const Close
+            = geometry::closure<OutputGeometry>::value == closed;
+        static bool const Reverse
+            = geometry::point_order<OutputGeometry>::value == counterclockwise;
+
+        // A hull for boxes is trivial. Any strategy is (currently) skipped.
+        boost::array<typename point_type<Box>::type, 4> range;
+        geometry::detail::assign_box_corners_oriented<Reverse>(box, range);
+        geometry::append(out, range);
+        if (BOOST_GEOMETRY_CONDITION(Close))
+        {
+            geometry::append(out, *boost::begin(range));
+        }
+    }
+};
+
+
+
+template <order_selector Order, closure_selector Closure>
+struct convex_hull_insert
+    : detail::convex_hull::hull_insert<Order, Closure>
+{};
+
+
+} // namespace dispatch
+#endif // DOXYGEN_NO_DISPATCH
+
+
+namespace resolve_strategy {
+
+struct convex_hull
+{
+    template <typename Geometry, typename OutputGeometry, typename Strategy>
+    static inline void apply(Geometry const& geometry,
+                             OutputGeometry& out,
+                             Strategy const& strategy)
+    {
+        //BOOST_CONCEPT_ASSERT( (geometry::concepts::ConvexHullStrategy<Strategy>) );
+        dispatch::convex_hull<Geometry>::apply(geometry, out, strategy);
+    }
+
+    template <typename Geometry, typename OutputGeometry>
+    static inline void apply(Geometry const& geometry,
+                             OutputGeometry& out,
+                             default_strategy)
+    {
+        typedef typename strategies::convex_hull::services::default_strategy
+            <
+                Geometry
+            >::type strategy_type;
+
+        apply(geometry, out, strategy_type());
+    }
+};
+
+struct convex_hull_insert
+{
+    template <typename Geometry, typename OutputIterator, typename Strategy>
+    static inline OutputIterator apply(Geometry const& geometry,
+                                       OutputIterator& out,
+                                       Strategy const& strategy)
+    {
+        //BOOST_CONCEPT_ASSERT( (geometry::concepts::ConvexHullStrategy<Strategy>) );
+
+        return dispatch::convex_hull_insert<
+                   geometry::point_order<Geometry>::value,
+                   geometry::closure<Geometry>::value
+               >::apply(geometry, out, strategy);
```

Comment:
This code takes the new umbrella strategy. There is no backward compatibility.

This is how it is done e.g. for `area` with `services::strategy_converter` "converting" from the old strategy to the new umbrella strategy:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/area.hpp#L240-L252
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/strategies/area/services.hpp#L42-L50
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/strategies/area/cartesian.hpp#L48-L54

Unless this PR breaks the backward compatibility anyway.

Reply:
This is a breaking change, the convex hull algorithm use to take an "agnostic" convex_hull strategy "graham_andrew" but now takes a new umbrella strategy to provides a side predicate.